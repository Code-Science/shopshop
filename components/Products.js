import React from 'react';
import PropTypes from 'prop-types';
import styles from '../styles/components/Products.module.css';
import Product from './Product';

function Products({ data }) {
  if (data.error) {
    return <p className={styles.error}>Something Went Wrong! Server is Down</p>;
  }
  const products = data.map((item) => <Product key={item.id} item={item} />);
  return (
    <div className={styles.products}>
      {products}
      {products}
    </div>
  );
}

Products.propTypes = {
  data: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.any),
    PropTypes.objectOf(PropTypes.any),
  ]),
};

export default Products;
