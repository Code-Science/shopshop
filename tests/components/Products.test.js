import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import React from 'react';
import Products from '../../components/Products';

configure({ adapter: new Adapter() });

describe('Products Component', () => {
  const mockItems = [
    {
      id: 4,
      product_name: 'shirt',
      price: 80,
      image1: 'image1',
      image2: 'image2',
    },
  ];
  const wrapper = shallow(<Products data={mockItems} />);

  it('expect to render products component', () => {
    expect(wrapper.length).toEqual(1);
  });
  it('snapshot test', () => {
    expect(wrapper).toMatchSnapshot();
  });
  it('Show error message if no items as props', () => {
    const mockErrorMessage = {
      error: 'error',
    };
    expect(shallow(<Products data={mockErrorMessage} />).text()).toEqual(
      'Something Went Wrong! Server is Down'
    );
  });
});
