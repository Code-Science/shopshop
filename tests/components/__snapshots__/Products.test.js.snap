// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Products Component snapshot test 1`] = `
<div
  className="products"
>
  <Product
    item={
      Object {
        "id": 4,
        "image1": "image1",
        "image2": "image2",
        "price": 80,
        "product_name": "shirt",
      }
    }
    key="4"
  />
  <Product
    item={
      Object {
        "id": 4,
        "image1": "image1",
        "image2": "image2",
        "price": 80,
        "product_name": "shirt",
      }
    }
    key="4"
  />
</div>
`;
